pkg/kubelet/certificate/certificate_manager.go

```diff
@@ -46,6 +46,9 @@ const (
 // manager. In the background it communicates with the API server to get new
 // certificates for certificates about to expire.
 type Manager interface {
+	// CertificateSigningRequestClient sets the client interface that is used for
+	// signing new certificates generated as part of rotation.
+	CertificateSigningRequestClient(certificatesclient.CertificateSigningRequestInterface)
```

Comment:
It's not used in this PR, but it's used in https://github.com/kubernetes/kubernetes/pull/41912. It's specifically to work around the one case where the API client needs the certificate from the certificate manager to make a connection, and the certificate manager needs the API client to request certificate changes. The sequence of steps is:

- Initialize certificate manager with local certs (either bootstrap or previously requested and stored cert). No client is passed in yet, so the certificate manager can't rotate/update yet.
- Initialize the API client with the certificate manager
- .Start() the certificate manager
    - certificate manager uses the client to request an updated certificate (if it needs too, either because of bootstrapping or expiration)